src/diagnostic/diagnostic_manager.h

```diff
@@ -143,6 +143,15 @@ DiagnosticsManager<Writer>::addDiagDict(PHARE::initializer::PHAREDict const& dia
     diagProps.computeTimestamps
         = diagInputs["compute_timestamps"].template to<std::vector<double>>();
 
+    diagProps.nAttributes = diagInputs["n_attributes"].template to<std::size_t>();
+    for (std::size_t i = 0; i < diagProps.nAttributes; ++i)
+    {
+        std::string idx = std::to_string(i);
+        std::string key = diagInputs["attribute_" + idx + "_key"].template to<std::string>();
+        std::string val = diagInputs["attribute_" + idx + "_value"].template to<std::string>();
```

Comment:
I was wondering if there was a way to just set the key to a dict node without need to extract it, maybe it's possible, in that case we wouldn't need to enforce all values to be strings